torch/_dynamo/variables/dicts.py

```diff
@@ -127,7 +137,7 @@ def __init__(self, vt) -> None:
             # TODO Temorarily remove to figure out what keys are we breaking on
             # and add proper support for them
             if not is_hashable(vt):
-                unimplemented(f"Dict key of type {type(vt)}. Key: {vt}")
+                raise_unhashable(vt)
```

Comment:
It makes more sense to keep the code as it is (raise `unimplemented`). But this gives a chance for Dynamo to capture and handle an exception when one tries to add an unhashable object to a set. If Dynamo doesn't capture, the exception is propagated to CPython.